clang/test/CXX/drs/cwg26xx.cpp

```diff
@@ -35,6 +35,29 @@ namespace std {
   template<typename T> T declval();
 } // namespace std
 
+namespace cwg2611 { // cwg2611: 21
+#if __cplusplus >= 201703L
+template<class> class R {};
+template<class... Ts> auto a(Ts... xs) -> R<decltype((xs + ...))>;
+template<class... Ts> auto b(Ts... xs) -> R<decltype((..., xs))>;
+template<class... Ts> auto c(Ts... xs, int i = 0) -> R<decltype((xs * ... * i))>;
+template<class... Ts> auto d(Ts... xs, int i = 0) -> R<decltype((i ^ ... ^ xs))>;
+R<int&> i = a(1);
+R<int&> j = b(1);
+R<int&> k = c();
+R<int&> l = d();
+
```

Comment:
I would add  `__is_same_as` tests as in the issue, that's the (only) tests we get wrong